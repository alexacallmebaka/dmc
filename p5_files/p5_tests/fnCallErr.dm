a : (x:int) int {
 return 5; 
}

b : (x:int) int { 
  a: int = 5; 
  return a;
}

c : (x:int) int {
 return true; 
}

d : (x:int) int {
 return x; 
}

g : (x:int, f:bool) int {
 return f; 
}

x : (y:int, f:bool) int {
 return;
}

y : (y:int, f:bool) void {
 return 5; 
}

z : (y:int, f:bool) int {
}

var1 : int;

main : () void {
  y(1, true);
  var1();
  g(true);
  d(true);
}