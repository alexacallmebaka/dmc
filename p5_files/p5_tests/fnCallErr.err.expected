FATAL [11,9]-[11,13]: Bad return value
FATAL [19,9]-[19,10]: Bad return value
FATAL [23,2]-[23,8]: Missing return value
FATAL [27,9]-[27,10]: Return with a value in void function
FATAL [38,3]-[38,7]: Attempt to call a non-function
FATAL [39,3]-[39,10]: Function call with wrong number of args
FATAL [40,5]-[40,9]: Type of actual does not match type of formal
FATAL [42,3]-[42,14]: Invalid assignment operation
Type Analysis Failed
